src/app/forms/validators/date/date.validators.ts

```diff
@@ -20,20 +20,26 @@ export class DateValidators {
       }
 
       if (includeTime) {
-        const [hours, minutes] = t.split(':');
-
-        if (!hours || !minutes) {
-          return { invalidDate: { error: true, reason: `'${label || 'Date'}' must include time` } };
-        }
-        if (23 < Number.parseInt(hours, 10)) {
-          return { invalidDate: { error: true, reason: `'${label || 'Date'}' hours must be between 0 and 23` } };
-        }
-        if (59 < Number.parseInt(minutes, 10)) {
-          return { invalidDate: { error: true, reason: `'${label || 'Date'}' minutes must be between 0 and 59` } };
-        }
+        return this.validateTime(t, label);
       }
 
       return null;
     };
   }
+
+  private static validateTime(time: string, label: string | undefined) {
+    const [hours, minutes] = time.split(':');
+
+    if (!hours || !minutes) {
+      return { invalidDate: { error: true, reason: `'${label || 'Date'}' must include time` } };
+    }
+    if (23 < Number.parseInt(hours, 10)) {
+      return { invalidDate: { error: true, reason: `'${label || 'Date'}' hours must be between 0 and 23` } };
+    }
+    if (59 < Number.parseInt(minutes, 10)) {
+      return { invalidDate: { error: true, reason: `'${label || 'Date'}' minutes must be between 0 and 59` } };
+    }
+
+    return null;
```

Comment:
should this method return null? when you have a return null on line 26

Reply:
I had a "not all code paths return a value" without it. If `includeTime` is true, we can just return `null` here, since the other `if`s before that call would have not been truthy. Without it, I'd probably have to do a check like:
```js
const validTime = this.validateTime(t,label)
if (validTime) {
  return validTime
}
```
And I obviously need the `null` on line 26 is needed in case `validateTime` is false

I do also like that this method is of type `ValidatorFn` which  is `any => ValidationErrors | null` (just like the `validDate` method on line 4), to keep it consistent with the rest of methods in the `custom-validators` folder. Maybe a return type on the function would make it more explicit.